vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+
+    @Value("${page_size}")
+    private int pageSize;
+
+    @RequestMapping(value = "/staff/search")
+    @GetMapping
+    public String getAllSearchedElementsWithoutNum(@RequestParam(name = "searchText") String searchString) {
+        return String.format("redirect:/staff/searchPage?page=%s&searchText=%s", "1", searchString);
+    }
+
+    @RequestMapping(value = "/staff/searchPage")
+    @GetMapping
+    public String getAllSearchedElements(@RequestParam(value = "page", required = false) String page, Model model,
+            @RequestParam(name = "searchText") String searchString,
+            @RequestParam(value = "tab", required = false) String tab) {
+
+        int spacePageNo;
+        int modulePageNo;
+        int slidePageNo;
+        int slideTextPageNo;
+        
+        if (tab == null || (tab != null && tab.trim().isEmpty())) {
+            tab = "module";
+        }
+        page = StringUtils.isEmpty(page) ? "1" : page;
+
+        int totalSpacePage = staffSearchManager.getCountOfSearchedSpace(searchString);
+        try {
+            spacePageNo = validatePageNumber(Integer.parseInt(page), totalSpacePage);
+        } catch (NumberFormatException numberFormatException) {
+            spacePageNo = 1;
+        }
+
+        int totalModulePage = staffSearchManager.getCountOfSearchedModule(searchString);
+        try {
+            modulePageNo = validatePageNumber(Integer.parseInt(page), totalModulePage);
+        } catch (NumberFormatException numberFormatException) {
+            modulePageNo = 1;
+        }
+
+        int totalSlidePage = staffSearchManager.getCountOfSearchedSlide(searchString);
+        try {
+            slidePageNo = validatePageNumber(Integer.parseInt(page), totalSlidePage);
+        } catch (NumberFormatException numberFormatException) {
+            slidePageNo = 1;
+        }
+
+        int totalSlideTextPage = staffSearchManager.getCountOfSearchedSlideText(searchString);
+        try {
+            slideTextPageNo = validatePageNumber(Integer.parseInt(page), totalSlideTextPage);
+        } catch (NumberFormatException numberFormatException) {
+            slideTextPageNo = 1;
+        }
+
+        // Reseting the page number of all other tabs apart from current tab to 1.
```

Comment:
why?

Reply:
apart from the current tab for all other three tab the page should be set to 1 otherwise if for the current tab let the page number is 3 then for the other tabs also page number coming as 3.